src/kysely.ts

```diff
@@ -1122,3 +1111,113 @@ export class Command<T> {
     return await this.#cb()
   }
 }
+
+function assertNotCommittedOrRolledBack(
+  state: ControllerTransctionState,
+): void {
+  if (state.isCommitted) {
+    throw new Error('Transaction is already committed')
+  }
+
+  if (state.isRolledBack) {
+    throw new Error('Transaction is already rolled back')
+  }
+}
+
+/**
+ * An executor wrapper that asserts that the transaction state is not committed
+ * or rolled back when a query is executed.
+ */
+class NotCommittedOrRolledBackAssertingExecutor implements QueryExecutor {
```

Comment:
should we mark it as @internal in the jsdoc?

Reply:
Done